Sources/TracingInstrumentation/Span.swift

```diff
@@ -121,34 +121,46 @@ public struct SpanAttributeKey<T>: Hashable, ExpressibleByStringLiteral {
 
 @dynamicMemberLookup
 public protocol SpanAttributeNamespace {
-    var attributes: SpanAttributes { get set }
+    /// Type that contains the nested attributes, e.g. HTTPAttributes which would contain `statusCode` and similar vars.
     associatedtype NestedAttributes: NestedSpanAttributesProtocol
+    
+    var attributes: SpanAttributes { get set }
 
-    subscript<T>(dynamicMember dynamicMember: KeyPath<SpanAttribute, SpanAttributeKey<T>>) -> SpanAttribute? { get set }
+    subscript<T>(dynamicMember dynamicMember: KeyPath<NestedAttributes, SpanAttributeKey<T>>) -> T? where T: SpanAttributeConvertible { get set }
 
     subscript<Namespace>(dynamicMember dynamicMember: KeyPath<SpanAttribute, Namespace>) -> Namespace
         where Namespace: SpanAttributeNamespace { get }
 }
 
 public protocol NestedSpanAttributesProtocol {
-    static var namespace: Self { get }
+    init()
+    static var __namespace: Self { get }
+}
+
+extension NestedSpanAttributesProtocol {
+    public static var __namespace: Self { .init() }
 }
 
 extension SpanAttributeNamespace {
-    public subscript<T>(dynamicMember dynamicMember: KeyPath<SpanAttribute, SpanAttributeKey<T>>) -> SpanAttribute? {
+    public subscript<T>(dynamicMember dynamicMember: KeyPath<NestedAttributes, SpanAttributeKey<T>>) -> T? where T: SpanAttributeConvertible {
         get {
-            let key = SpanAttribute.__magic[keyPath: dynamicMember]
-            return self.attributes[key.name]
+            let key = NestedAttributes.__namespace[keyPath: dynamicMember]
+            let spanAttribute: SpanAttribute? = self.attributes[key.name]
+            if let value = spanAttribute?.anyValue {
+                return value as? T
```

Comment:
some nasty casting here to be able to implement the subscript in a type safe way